### PR TITLE
[Data] [Docs] Add WebDataset to docs

### PR DESCRIPTION
**Pull request by amogkam**
Adds `WebDatasetDatasource` to the docs

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/api/input_output.rst [hunk 1]
```diff
@@ -226,6 +226,7 @@ Built-in Datasources
    datasource.RangeDatasource
    datasource.TFRecordDatasource
    datasource.MongoDatasource
+   datasource.WebDatasetDatasource
```
Comment (c21) [inline doc/source/data/api/input_output.rst:238]: can you also add a section of `WebDatasetDatasource` for `read_webdataset`? 

Example of image - https://github.com/ray-project/ray/blob/master/doc/source/data/api/input_output.rst?plain=1#L62-L68 .

File: doc/source/data/api/input_output.rst [hunk 2]
```diff
@@ -195,6 +195,15 @@ TensorFlow
 
    from_tf
 
+
```
Comment (c21) [inline doc/source/data/api/input_output.rst:198]: nit: unnecessary extra line.

File: doc/source/data/api/input_output.rst (hunk 1 above)
Comment (amogkam) [reply doc/source/data/api/input_output.rst:238]: added

File: doc/source/data/api/input_output.rst (hunk 2 above)
Comment (amogkam) [inline doc/source/data/api/input_output.rst:198]:
```suggestion
```

File: doc/source/_toc.yml [hunk 3]
```diff
@@ -85,6 +85,8 @@ parts:
       - file: data/dataset
         title: Ray Data
         sections:
+          - file: data/dataset
+            title: "What is Ray Data?"
```
Comment (c21) [inline doc/source/_toc.yml:89]: should we keep consistency across all libraries? Why this is a problem for Ray Data only?

Comment (amogkam) [reply doc/source/_toc.yml:89]: It's a problem for all libraries, but will let each library owner decide what they want to do. Didn't want to overload everything in a single PR.

**Review by c21 — APPROVED** [review]
Looks ok to me.

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging to master

**Pull request merged**